### ISSUE TITLE
Unable to fetch or push data packages

### ISSUE THREAD

**Issue by v12venator** [brian7704/OpenTAKServer#39]
When connected to an opentakserver, EUD is unable to push or query data packages to/from server. Found all network connectivity settings and ensured the device was able to connect to 8443 on the server through the NGINX proxy. The EUD's return code for data package query is:

Data Package Search Failed
Socket is closed

Attempting to run this from ATAK client on two different Debian 12 opentakservers, one with Let's Encrypt certificates and one with the generated self-signed certificates.

**Comment by v12venator** [comment]
Looked at the debug nginx log again, and the proxy is returning SSL error 46 when a query attempt is made

**Comment by brian7704** [comment]
I'm guessing that error is from the server with the self signed cert. Have you loaded truststore-root.p12 into ATAK in the server config?

**Comment by v12venator** [comment]
Yes, I loaded the p12 file into the truststore slot when configuring and enrolling the device with the server, and for the self-signed server, I also loaded the self signed CA public certificate into the android system store. I also set up a third and fourth ubuntu and ubuntu server installation separately with self-signed certificates and the issue did not persist on those server.

I suspect that the root cause is a bug with the Debian provided version of nginx, where the ssl_client_certificate directive in the ots_https config file is not truly honored. I will be attempting to replace the Debian version with the Nginx provided mainline and stable versions to see if the issue persists or is resolved.

**Comment by v12venator** [comment]
I was able to test both the mainline and stable versions of Nginx on Debian 12, and the behavior was observed with both versions from the Nginx repository for the software.

**Comment by brian7704** [comment]
Can you post your nginx configs?

**Comment by v12venator** [comment]
My apologies for misunderstanding, I was not using the config bundle to enroll the device, I was entering all of the info manually to connect to the server.

The relevant configs are attached.
[ots_https.txt](https://github.com/user-attachments/files/16112784/ots_https.txt)
[ots_certificate_enrollment.txt](https://github.com/user-attachments/files/16112786/ots_certificate_enrollment.txt)

I reviewed the example docs on the io site, and I don't see any discrepancies with the attached configs, aside from the cert locations, but the OTS directory is owned by the unprivileged user that was used to install the server.

I also installed the CA public certificate to the Debian system trust store with dpkg-reconfigure to see if that would fix the unknown CA error codes that the NGINX server was reporting, but that did not make any difference after import.

**Comment by v12venator** [comment]
Good evening Brian,

I've decided that running the OpenTAKServer on a Debian 12 server requires too much intervention to make it work, and I still have not found a solution to the Debian installation of Nginx not honoring the CA file configuration.

Additionally, I had to pass the --break-system-packages switch to the "pip install opentakserver" line, which I'm sure has unseen configuration issues that I haven't seen yet.

As such, I've deprovisioned the Debian 12 server and replaced it with an Ubuntu 23.04 system.

Thank you for putting time towards this issue, and perhaps this can be used as an example for why OTS is not supported on a Debian system.
